### ISSUE TITLE
TEMPLATE_SCENARIO

### ISSUE THREAD

**Issue by HazardyKnusperkeks** [catchorg/Catch2#1990]
**Description**
It would be nice to use templated Code when using BDD.

**Comment by arcondello** [comment]
I had the same request, so documenting my workaround in case others come to the same place.

In 1.x and 2.x SCENARIO is a just a wrapper for TESTCASE https://github.com/catchorg/Catch2/blob/18ab353e5580d5d2558bacab5a68d7f6f40c8a79/include/catch.hpp#L292

so 

```
TEMPLATE_TEST_CASE( "Scenario: vectors can be sized and resized", "[vector][template]", int, std::string, (std::tuple<int,float>) ) {

    std::vector<TestType> v( 5 );

    GIVEN( "A vector with some items" ) {
        std::vector<int> v( 5 );

        REQUIRE( v.size() == 5 );
        REQUIRE( v.capacity() >= 5 );

        WHEN( "the size is increased" ) {
            v.resize( 10 );

            THEN( "the size and capacity change" ) {
                REQUIRE( v.size() == 10 );
                REQUIRE( v.capacity() >= 10 );
            }
        }

    ...
```
seems to work just fine for me.

It does appear that in 3.x things get more complicated
https://github.com/catchorg/Catch2/blob/ce54ec185f69e32661efdcb4659f892415316b0e/src/catch2/catch_test_macros.hpp#L194

**Comment by HazardyKnusperkeks** [comment]
Yeah, I always wanted to create a pull request for that, but did not find the time. Maybe I will still do it this year.
I don't know which code is selected when in 3.x, but
https://github.com/catchorg/Catch2/blob/ce54ec185f69e32661efdcb4659f892415316b0e/src/catch2/catch_test_macros.hpp#L152
is still there.
